frontendActions/src/main/java/com/virtuslab/gitmachete/frontend/actions/SlideOutSelectedBranchAction.java

```diff
@@ -0,0 +1,61 @@
+package com.virtuslab.gitmachete.frontend.actions;
+
+import static com.virtuslab.gitmachete.frontend.actions.ActionsUtils.getSelectedMacheteBranch;
+
+import com.intellij.openapi.actionSystem.AnActionEvent;
+import com.intellij.openapi.actionSystem.CommonDataKeys;
+import io.vavr.control.Option;
+import org.checkerframework.checker.guieffect.qual.UIEffect;
+
+import com.virtuslab.gitmachete.backend.api.BaseGitMacheteBranch;
+import com.virtuslab.gitmachete.backend.api.IGitMacheteRepository;
+import com.virtuslab.gitmachete.frontend.keys.DataKeys;
+
+/**
+ * Expects DataKeys:
+ * <ul>
+ *  <li>{@link DataKeys#KEY_BRANCH_LAYOUT}</li>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_FILE_PATH}</li>
+ *  <li>{@link DataKeys#KEY_GIT_MACHETE_REPOSITORY}</li>
+ *  <li>{@link DataKeys#KEY_IS_GIT_MACHETE_REPOSITORY_READY}</li>
+ *  <li>{@link DataKeys#KEY_SELECTED_BRANCH_NAME}</li>
+ *  <li>{@link CommonDataKeys#PROJECT}</li>
+ * </ul>
+ */
+public class SlideOutSelectedBranchAction extends BaseSlideOutBranchAction {
+
+  @Override
+  @UIEffect
+  public void update(AnActionEvent anActionEvent) {
+    super.update(anActionEvent);
+
+    var presentation = anActionEvent.getPresentation();
+    if (presentation.isVisible()) {
+      Option<BaseGitMacheteBranch> selectedBranch = getSelectedMacheteBranch(anActionEvent);
+      if (selectedBranch.isDefined()) {
```

Comment:
Oh good point (the same for rebasing selected branch). Actually these actions are called directly after selecting some branch but it will be better to vanish & disable them here.